runtime/src/installed_scheduler_pool.rs

```diff
@@ -50,10 +126,59 @@ pub trait InstalledScheduler: Send + Sync + Debug + 'static {
     /// two reasons later.
     #[must_use]
     fn wait_for_termination(&mut self, reason: &WaitReason) -> Option<ResultWithTimings>;
+
+    fn context<'a>(&'a self) -> Option<&'a SchedulingContext>;
+    fn replace_context(&mut self, context: SchedulingContext);
 }
 
 pub type DefaultInstalledSchedulerBox = Box<dyn InstalledScheduler>;
 
+pub type InstalledSchedulerPoolArc = Arc<dyn InstalledSchedulerPool>;
+
+pub type SchedulerId = u64;
+
+/// A small context to propagate a bank and its scheduling mode to the scheduler subsystem.
+///
+/// Note that this isn't called `SchedulerContext` because the contexts aren't associated with
+/// schedulers one by one. A scheduler will use many SchedulingContexts during its lifetime.
+/// "Scheduling" part of the context name refers to an abstract slice of time to schedule and
+/// execute all transactions for a given bank for block verification or production. A context is
+/// expected to be used by a particular scheduler only for that duration of the time and to be
+/// disposed by the scheduler. Then, the scheduler may work on different banks with new
+/// `SchedulingContext`s.
+#[derive(Clone, Debug)]
+pub struct SchedulingContext {
+    // mode: SchedulingMode, // this will be added later.
+    bank: Arc<Bank>,
+}
+
+impl SchedulingContext {
+    pub fn new(bank: Arc<Bank>) -> Self {
+        Self { bank }
+    }
+
+    pub fn bank(&self) -> &Arc<Bank> {
+        &self.bank
+    }
+
+    pub fn slot(&self) -> Slot {
+        self.bank().slot()
+    }
+
+    pub fn log_prefix(context: Option<&Self>, scheduler_id: SchedulerId) -> String {
```

Comment:
What is this for? I don't see any non-test usage in the main PR either.
Do you intend to actually use this? It seems like re-creating a String everytime we want to log from scheduler will be slow

Reply:
> What is this for? I don't see any non-test usage in the main PR either.

my bad. yeah, it'll be used by the real unifified scheduler logic, which isn't in the main pr.. for now just removed: 0fb76641d5c975fd6b37a0573009a6ce50117fbe

Reply:
> Do you intend to actually use this? It seems like re-creating a String everytime we want to log from scheduler will be slow

fyi, this will be addressed when i re-add this. thanks!